lib/PushAssociations.js

```diff
@@ -85,7 +85,7 @@ var removeForUser = function (user, callback) {
 };
 
 var removeDevice = function (token, callback) {
-    PushAssociation.remove({token: token}, function (err) {
+    PushAssociation.deleteMany({token: token}, function (err) {
```

Comment:
@campbza @idiomattic since both you were dealing with this recently - can you confirm if the error handling will actually work here? From what I understand the `deleteMany` call has to be rewritten to be:

```
model.deleteMany(condtiions).then(successCase).catch(errorHandling)
```
to properly handle errors.

Reply:
it can indeed take a callback as the final argument: https://mongoosejs.com/docs/6.x/docs/api/query.html#query_Query-deleteMany

This works whether or not an options map is provided.

Reply:
I can just use deprecated functions since we close to sunset with this service. Which problems with moving off deprecated functions you guys had?

Reply:
@campbza and I were looking into error handling in `deleteMany` when adding tests for it, and the code had to be rewritten like this:

https://github.com/peerspace/ListingsAPI/pull/374/files#diff-450ed0e0d1bc4ba81ab72aad6d4c994e98b4f0f714da868515f41981d4e5afe8R263

then again, we only run into this when adding unit tests - since it made sense to ensure invalid inputs can be processed

Reply:
Got it. Thanks @lukaszkorecki-ps 
I've restored deprecated functions